### ISSUE TITLE
The metaserver service exits by itself

### ISSUE THREAD

**Issue by YunhuiChen** [opencurve/curve#1135]
**Describe the bug (描述bug)**
The metaserver service exits by itself

**To Reproduce (复现方法)**
1、stop one metaserver  a99eebe453e5
2、other metaservers exits by itself
Get Service Status: [OK]  

cluster name    : citest
cluster kind    : curvefs
cluster mds addr: 10.182.26.16:6700,10.182.26.17:6700,10.182.26.18:6700

Id            Role        Host          Replica  Container Id  Status                   
--            ----        ----          -------  ------------  ------                   
9e5192e5c4eb  etcd        10.182.26.16  1/1      816fa23a6195  Up 24 hours              
8f880895df33  etcd        10.182.26.17  1/1      1403a8f69a0c  Up 24 hours              
a543f5441ccd  etcd        10.182.26.18  1/1      45b47979cbb2  Up 24 hours              
40747c90f20f  mds         10.182.26.16  1/1      fba768f426a0  Up 24 hours              
f333dff92093  mds         10.182.26.17  1/1      5ef682428faf  Up 24 hours              
6852f760b287  mds         10.182.26.18  1/1      cbbe78ef8c0b  Up 24 hours              
fc9bf1e50ef3  metaserver  10.182.26.16  1/1      f92f2fc90938  Exited (134) 17 hours ago
a99eebe453e5  metaserver  10.182.26.16  1/1      bbb62194e359  Exited (137) 22 hours ago
a6db1620f680  metaserver  10.182.26.17  1/1      92d1bb11751a  Up 24 hours              
51a1e0f81094  metaserver  10.182.26.17  1/1      37f31b486cc7  Up 24 hours              
3772ae3c2237  metaserver  10.182.26.18  1/1      bf08fbda1e61  Up 25 minutes            
caaaa38920b5  metaserver  10.182.26.18  1/1      71fe03f525e7  Exited (134) 20 hours ago

![image](https://user-images.githubusercontent.com/67582932/156716507-5874af0e-2427-4ce0-aea3-bbd90d2f3369.png)

**Expected behavior (期望行为)**

**Versions (各种版本)**
OS:
Compiler:
curve-mds:
curve-chunkserver:
curve-snapshotcloneserver:
curve-sdk:
nebd:
curve-nbd:

**Additional context/screenshots (更多上下文/截图)**

**Comment by h0hmj** [comment]
substr out_of_range 

add some log and try to reproduce/locate problem.

**Comment by h0hmj** [comment]
```
diff --git a/curvefs/src/metaserver/s3compact_wq_impl.cpp b/curvefs/src/metaserver/s3compact_wq_impl.cpp
index 3398a82c..7abbb152 100644
--- a/curvefs/src/metaserver/s3compact_wq_impl.cpp
+++ b/curvefs/src/metaserver/s3compact_wq_impl.cpp
@@ -358,6 +358,19 @@ int S3CompactWorkQueueImpl::ReadFullChunk(
                 continue;
             }
             for (const auto& req : reqs) {
+                if(buf.length() < req->off+req->len){
+                    LOG(ERROR) << "debug s3compact: objname: " << objName
+                        << ", objsize: " << buf.length()
+                        << ", req_off: " << req->off
+                        << ", req_len: " << req->len;
+                    LOG(ERROR) << "s3compact: s3 request generated";
+                    for (const auto& s3req : s3reqs) {
+                        LOG(ERROR) << "index:" << s3req.reqIndex << ", zero:" << s3req.zero
+                                << ", s3objname:" << s3req.objName << ", off:" << s3req.off
+                                << ", len:" << s3req.len;
+                    }
+                    return -1;
+                }
                 readContent[req->reqIndex] = buf.substr(req->off, req->len);
             }
             break;
@@ -539,6 +552,15 @@ void S3CompactWorkQueueImpl::CompactChunk(
     int ret = ReadFullChunk(compactCtx, validList, &fullChunk, &newChunkInfo);
     if (ret != 0) {
         LOG(WARNING) << "s3compact: ReadFullChunk failed, index " << index;
+        for (auto i = 0; i < s3chunkinfolist.s3chunks_size(); i++) {
+            auto chunkinfo = s3chunkinfolist.s3chunks(i);
+            // print s3chunkinfolist
+            LOG(ERROR) << "chunkid: " << chunkinfo.chunkid()
+                    << ", offset:" << chunkinfo.offset()
+                    << ", len:" << chunkinfo.len()
+                    << ", compaction:" << chunkinfo.compaction()
+                    << ", zero: " << chunkinfo.zero();
+        }
         s3infoCache_->InvalidateS3Info(
             compactCtx.fsId);  // maybe s3info changed?
         return;
```

**Comment by h0hmj** [comment]
read failed. s3 object not exist.
root cause:
fs created using blocksize 1M, chunksize 4M
client rw using blocksize 4M, chunksize 64M

**Comment by h0hmj** [comment]
may be fixed by 780ded6b13f9e9ee772dbed2cb891ff284074fc6
